### ISSUE TITLE
Parallel HDF5?

### ISSUE THREAD

**Issue by eschnett** [JuliaPackaging/Yggdrasil#2857]
The HDF5 library can be built in a parallel mode. In this mode, it has MPI as dependency, and allows parallel access to data files.

I wonder whether/how this could be supported. One option would be to always require MPI as dependency, as the parallel build offers (I think) a superset of the serial functionality. Of course, pulling in MPI as dependency might be inconvenient for other packages.

The other option would be to have two versions of HDF5, serial and parallel. As long as only the serial version is required, it can be used; if both the serial and parallel versions are required, then this should be satisfied by the parallel version.

**Comment by giordano** [comment]
Yeah, this is a general problem we have at the moment: offering alternative builds of packages with different options, which is rather simple to do with Spack instead.  The main problem for us is that `Pkg` doesn't support optional or alternative packages.  The best think you can do at the moment is to build a different package (i.e., different name) with the extra feature you want (we have some `*_CUDA` packages for example).

Note that for the specific case of HDF5, we don't build it all because it doesn't cross-compile, we simply redistribute Conda binaries.

**Comment by ViralBShah** [comment]
I believe we now build HDF5 with MPI.
